src/dune_rules/scope.ml

```diff
@@ -61,24 +61,31 @@ module DB = struct
           | Library (_, { project; public = Some p; _ }) ->
             Some (Dune_file.Public_lib.name p, Project project)
           | Library _ -> None
-          | Deprecated_library_name
-              { old_public_name = { public = old_public_name; _ }
-              ; new_public_name
-              ; _
-              } ->
-            Some
-              (Dune_file.Public_lib.name old_public_name, Name new_public_name))
+          | Library_redirect s ->
+            let old_name =
```

Comment:
I'm not sure this is correct. Do you understand why it's correct to put a local name into a table keyed on public names?

Note that below (in the collision-handling branch) you *do not* take `Library_redirect` into account, so I expect `assert false` to actually be reachable by having multiple public libraries with the same name.

Reply:
Good call, I'm contaminating the wrong scope with these private names. The public library scope is shared between all scopes so it shouldn't be touched. I pushed a commit to fix this.